tests/lib/connectors/collect.ts

```diff
@@ -0,0 +1,131 @@
+/* eslint-disable no-sync */
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as url from 'url';
+import * as sinon from 'sinon';
+
+import test from 'ava';
+
+import { builders } from '../../helpers/connectors';
+import { createServer } from '../../helpers/test-server';
+import { IConnector, IConnectorBuilder, INetworkData } from '../../../src/lib/types'; // eslint-disable-line no-unused-vars
+import { generateHTMLPage } from '../../helpers/misc';
+
+test.beforeEach(async (t) => {
+    const sonar = {
+        emit() { },
+        emitAsync() { }
+    };
+
+    const server = createServer();
+
+    await server.start();
+
+    sinon.spy(sonar, 'emit');
+    sinon.spy(sonar, 'emitAsync');
+
+    t.context = {
+        server,
+        sonar
+    };
+});
+
+test.afterEach.always((t) => {
+    t.context.server.stop();
+    t.context.sonar.emit.restore();
+    t.context.sonar.emitAsync.restore();
+});
+
+const pathToFaviconInDir = path.join(__dirname, './fixtures/common/favicon.ico');
+const pathToFaviconInLinkElement = path.join(__dirname, './fixtures/common/favicon-32x32.png');
+
+const runTest = async (t, connectorBuilder, serverConfig?) => {
+    const { sonar } = t.context;
+    const connector: IConnector = await (connectorBuilder)(sonar, {});
+    const server = t.context.server;
+
+    if (serverConfig) {
+        server.configure(serverConfig);
+    }
+
+    await connector.collect(url.parse(`http://localhost:${server.port}/`));
+    await connector.close();
+};
+
+const testConnectorCollect = (connectorInfo) => {
+    const connectorBuilder: IConnectorBuilder = connectorInfo.builder;
+    const name: string = connectorInfo.name;
+
+    test.serial(`[${name}] Favicon is present in a 'link' element with 'rel' attribute set to 'icon' `, async (t) => {
+        const faviconInLinkElementDir = `http://localhost:${t.context.server.port}/images/favicon-32x32.png`;
+        const serverConfig = {
+            '/': generateHTMLPage(`<link rel="icon" type="image/png" href="/images/favicon-32x32.png" sizes="32x32">`),
+            '/images/favicon-favicon-32x32.png': fs.readFileSync(pathToFaviconInLinkElement)
+        };
+
+        await runTest(t, connectorBuilder, serverConfig);
+
+        t.plan(2);
```

Comment:
`t.plan` is not necessary here. Take a look to [this](https://github.com/avajs/ava/blob/master/docs/recipes/when-to-use-plan.md)

It makes sense when you need to be sure that a block of code with `t.is` (or another test) is run.
For example:
```
t.plan(2);
try {
    await runTest(t, connectorBuilder, serverConfig);
    t.is(t.context.sonar.emitAsync.withArgs('fetch::end').callCount, 1);
    t.is(t.context.sonar.emitAsync.withArgs('fetch::end').args[0][1].request.url, faviconInLinkElementDir);
}catch(e){}
```